aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -140,6 +140,41 @@ bool is_nonzero(const Tensor& self) {
   TORCH_INTERNAL_ASSERT(false, "Expected non-Tensor backend scalar");
 }
 
+namespace {
+
+inline at::Tensor scalar_to_tensor_default_dtype(
+    Scalar s,
+    const Device device = at::kCPU) {
```

Comment:
Better not include a default argument here for clarity. Default arguments are best used for user-facing APIs, but this will be developer-facing.